app/src/main/java/ch/epfl/cs311/wanderwave/model/spotify/SpotifyController.kt

```diff
@@ -305,6 +307,87 @@ constructor(
     }
   }
 
+  /**
+   * Retrieve the image URL of an album.
+   *
+   * @param albumId The Spotify ID of the album.
+   * @return A Flow that emits the album image URL as a string.
+   */
+  fun getAlbumImage(albumId: String): Flow<String?> {
+    return callbackFlow {
+      val accessToken = authenticationController.getAccessToken()
+
+      if (accessToken == null) {
+        trySend(null)
+        close()
+        return@callbackFlow
+      }
+
+      val urlString = "https://api.spotify.com/v1/albums/$albumId"
+      val response = authenticationController.makeApiRequest(URL(urlString))
+
+      withContext(Dispatchers.IO) {
+        try {
```

Comment:
Here, can you just have one trySend(null), like
if(!failure){
....
if(image lenght){
trySend image
}
trySend(null)
}

it can help with reducing the complexity